guides/source/active_model_basics.md

```diff
@@ -38,82 +67,385 @@ class EmailContact
+`ActiveModel::Attributes`.
+
+#### Method: `attribute_names`
+
+The `attribute_names` method returns an array of attribute names.
+
+```irb
+irb> Person.attribute_names
+=> ["name", "date_of_birth", "active"]
+```
+
+#### Method: `attributes`
+
+The `attributes` method returns a hash of all the attributes with their names as
+keys and the values of the attributes as values.
+
+```irb
+irb> person.attributes
+=> {"name" => "Jane", "date_of_birth" => Wed, 01 Jan 2020, "active" => false}
+```
+
+### Attribute Assignment
+
+[`ActiveModel::AttributeAssignment`](https://api.rubyonrails.org/classes/ActiveModel/AttributeAssignment.html)
+allows you to set an object's attributes by passing in a hash of attributes with
+keys matching the attribute names. This is useful when you want to set multiple
+attributes at once.
+
+Consider the following class:
+
+```ruby
+class Person
+  include ActiveModel::AttributeAssignment
+
+  attr_accessor :name, :date_of_birth, :active
+end
+```
+
+```irb
+irb> person = Person.new
+
+# Set multiple attributes at once
+irb> person.assign_attributes(name: "John", date_of_birth: "1998-01-01", active: false)
+
+irb> person.name
+=> "John"
+irb> person.date_of_birth
+=> Thu, 01 Jan 1998
+irb> person.active
+=> false
+```
+
+If the passed hash responds to the `permitted?` method and the return value of
+this method is `false`, an `ActiveModel::ForbiddenAttributesError` exception is
+raised.
+
+NOTE: `permitted?` is used for [strong
+params](https://guides.rubyonrails.org/action_controller_overview.html#strong-parameters)
+integration whereby you are assigning a params attribute from a request.
+
+```irb
+irb> person = Person.new
+
+# Using strong parameters checks, build a hash of attributes similar to params from a request
+irb> params = ActionController::Parameters.new(name: "John")
+=> #<ActionController::Parameters {"name" => "John"} permitted: false>
+
+irb> person.assign_attributes(params)
+=> # Raises ActiveModel::ForbiddenAttributesError
+irb> person.name
+=> nil
+
+# Permit the attributes we want to allow assignment
+irb> permitted_params = params.permit(:name)
+=> #<ActionController::Parameters {"name" => "John"} permitted: true>
+
+irb> person.assign_attributes(permitted_params)
+irb> person.name
+=> "John"
+```
+
+#### Method alias: `attributes=`
+
+The `assign_attributes` method has an alias `attributes=`.
+
+INFO: A method alias is a method that performs the same action as another
+method, but is called something different. Aliases exist for the sake of
+readability and convenience.
+
+The following example demonstrates the use of the `attributes=` method to set
+multiple attributes at once:
+
+```irb
+irb> person = Person.new
+
+irb> person.attributes = { name: "John", date_of_birth: "1998-01-01", active: false }
+
+irb> person.name
+=> "John"
+irb> person.date_of_birth
+=> "1998-01-01"
+```
+
+##### Why does `assign_attributes` use `()` but `attributes=` use `{}` to pass the same attributes?
```

Comment:
Based on [this conversation](https://github.com/Ridhwana/rails/pull/2#discussion_r1497379173), I've added this section. I considered adding to a miscellaneous section at the bottom of this guide but it felt too far removed from the original context. I would have possibly preferred to put this in an INFO (albeit it may be too long for an info box), however, I could not get the syntax quite right to include the code samples. 

If anyone has thoughts ot feedback, feel free to drop them here.

Reply:
This may be over-rotating on on explanation but it's definitely something that I feel a beginner would find useful. Although reading this, I still find it a little confusing, so if possible slimming it down would be better.

Would be curious to hear what the other 2 think of this level of explanation. 

And it's a pity we can't fit it into an info box, that would make much more sense. I don't suppose there's a way to explain without the additional code snippets?

Reply:
It's tricky to find the right balance of how much detail to give, because we are mostly explaining Ruby syntax at this point. How much of that we need to touch may vary on a case-by-case basis, but in general I don't think we need to get into every single bit of the Ruby language.

Here's a potential suggestion to shorten that to an info box:

```
INFO: `assign_attributes` and `attributes=` are both method calls, and accept
the hash of attributes to assign as argument. In many cases Ruby allows parens
`()` from method calls, and curly braces `{}` from hash definitions, to be
omitted. "Setter" methods like `attributes=` are commonly written without `()`,
even though including them works the same, and they require the hash to always
include `{}`: `person.attributes=({ name: "John" })` is fine, but
`person.attributes = name: "John"` results in a `SyntaxError`. Other method
calls like `assign_attributes` may or may not contain both parens `()` and `{}`
for the hash argument. For example, `assign_attributes name: "John"` and
`assign_attributes({ name: "John" })` are both perfectly valid Ruby code, however
`assign_attributes { name: "John" }` is not, because Ruby can't differentiate
that hash argument from a block, and will raise a `SyntaxError`.
```

![Screenshot 2024-02-26 at 10 55 40](https://github.com/Ridhwana/rails/assets/26328/4f68f00a-08e6-4412-8e47-74e5927322dd)

I tried to include some examples within the paragraph, there's no need to show the output but I think at least show one or two examples of what is valid and sometimes invalid adds to the explanation. We might be able to omit a bit more if that needs to be shortened further.

Reply:
Thank you for taking a stab at this Carlos! I think it reads really well! I know that we're trying to shorten it, but I find myself being overwhelmed by the single long paragraph and so I added some line breaks in the middle:

<img width="675" alt="Screenshot 2024-02-26 at 23 37 09" src="https://github.com/Ridhwana/rails/assets/2786819/335d149f-91ee-4d19-890e-145e20879c78">

Reply:
Looks great!